### PR TITLE
RFC: Adding 101 demo

### PR DESCRIPTION
**Pull request by fao89**
Ref: https://discourse.pulpproject.org/t/rfc-pulp-101-webpage/155/8

### CODE REVIEW

**Comment by fao89** [comment]
@melcorr

File: .github/workflows/demo.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                      --volume "/$(pwd)/containers":/var/lib/containers:Z \
+                      --device /dev/fuse \
+                      pulp/pulp:https
+
+            # Wait server to be up
+            sleep 10
+            for _ in $(seq 20)
+            do
+              sleep 2
+              if curl --insecure --fail https://localhost/pulp/api/v3/status/ > /dev/null 2>&1
```
Comment (fao89) [inline .github/workflows/demo.yml:39]: This is the step where we set the Pulp server.
The other steps are client steps

File: .github/workflows/demo.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            sudo update-ca-certificates
+
+            # Setting the password for admin
+            docker exec pulp bash -c 'pulpcore-manager reset-admin-password --password ci_password'
+
+      - name: Setting Pulp CLI
+        run: |
+          pip install pulp-cli[pygments]
+          pulp config create --username admin --base-url https://pulp.example.com --password ci_password
+          pulp status
```
Comment (fao89) [inline .github/workflows/demo.yml:69]: Starting from here, these are the client steps.

Let's split the client steps into 3 types of users:
- admin
- regular user
- QE

Here we are setting the CLI for the admin, the person who will create remotes/repositories/publications/distributions/...

File: .github/workflows/demo.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          echo "PUBLICATION_HREF=$PUBLICATION_HREF" >> $GITHUB_ENV
+
+      - name: Distribute "dev" packages
+        run: |
+          pulp python distribution create --name "dev" --base-path "dev" --publication $PUBLICATION_HREF
+
+      - name: Publish & Distribute "prod" packages
+        run: |
+          PROD_PUBLICATION_HREF=$(pulp python publication create --repository "prod" | jq -r .pulp_href)
+          pulp python distribution create --name "prod" --base-path "prod" --publication $PROD_PUBLICATION_HREF
```
Comment (fao89) [inline .github/workflows/demo.yml:101]: Client-side - where admin user is prepping pulp to be used by QE/regular users

File: .github/workflows/demo.yml [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+      - name: Install package from pulp "dev" repository
+        run: |
+          pip install -vvv --trusted-host pulp -i https://pulp.example.com/pypi/dev/simple/ shelf-reader
```
Comment (fao89) [inline .github/workflows/demo.yml:105]: Client-side, QE user testing dev content

(consider pip installing from dev distribution as a test)

Here I could explore more and add some tests

File: .github/workflows/demo.yml [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+
+      - name: Sync packages from "dev" to "prod"
+        run: |
+          pulp python repository sync --name "prod" --remote "prod"
```
Comment (fao89) [inline .github/workflows/demo.yml:109]: Client-side, QE user

Here is the part where the content is "promoted" from dev to prod repository

Maybe I should start another Pulp CLI with QE user?

File: .github/workflows/demo.yml [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+      - name: Install package from pulp "prod" repository
+        run: |
+          pip install -vvv --force-reinstall --trusted-host pulp -i https://pulp.example.com/pypi/prod/simple/ shelf-reader
```
Comment (fao89) [inline .github/workflows/demo.yml:113]: Client-side, regular user

User pip installing python packages from pulp prod distribution

File: .github/workflows/demo.yml (hunk 3 above)
Comment (melcorr) [reply .github/workflows/demo.yml:101]: Is it realistic that this is part of the same GHA script?

File: .github/workflows/demo.yml (hunk 4 above)
Comment (melcorr) [reply .github/workflows/demo.yml:105]: Yeah that is what I think we mean when we say CI/CD scenario. 
Integration tests when a dev pushes changes at this point. 
Sorry for making work for you @fao89  :(

Comment (fao89) [reply .github/workflows/demo.yml:105]: this is a PoC, we are learning the best way to communicate with users. We'll probably have a lot more changes!

**Pull request closed without merging**